fdbserver/ClusterController.actor.cpp

```diff
@@ -3045,6 +3021,58 @@ ACTOR Future<Void> monitorRatekeeper(ClusterControllerData *self) {
 	}
 }
 
+ACTOR Future<Void> dbInfoUpdater( ClusterControllerData* self ) {
+	state Future<Void> dbInfoChange = self->db.serverInfo->onChange();
+	state Future<Void> updateDBInfo = self->updateDBInfo.onTrigger();
+	loop {
+		choose {
+			when(wait(updateDBInfo)) {
+				wait(delay(SERVER_KNOBS->DBINFO_BATCH_DELAY) || dbInfoChange);
+			}
+			when(wait(dbInfoChange)) {}
+		}
+		
+		if(dbInfoChange.isReady()) {
+			self->updateDBInfoEndpoints.clear();
+			for(auto &it : self->id_worker) {
+				self->updateDBInfoEndpoints.push_back(it.second.details.interf.updateServerDBInfo.getEndpoint());
+			}
+		} else {
+			uniquify(self->updateDBInfoEndpoints);
```

Comment:
Could `updateDBInfoEndpoints` be a set?

Reply:
I believe it should be slightly more efficient as a vector, one sort should be less expensive than a tree insertion for every element

Reply:
Possibly, though it should be asymptotically the same. Are you expecting this to be a costly thing?